CS Implement

TomChermin opened the pull request:
Added functionality for CS
I think it works now and no longer has negative bounty possibilities

TomChermin committed:
CS Implement

Added functionality for CS

MaikoKingma commented on Strings.ts:
```diff
@@ -177,4 +179,19 @@ export class Strings {
         return `📃 @${bountyPlacer} put a bounty worth ${bounty} points on @${bountyTarget} !`;
     }
+
+    static currentlyCSuserName: string | null = null): string {
```
missing an ( and no parameter name

MaikoKingma commented on Strings.ts:
```diff
@@ -132,6 +133,7 @@ export class Strings {
         + `/${Commands.placebounty} - Place a bounty on a player\n`
         + `/${Commands.kill} - Attempt to kill a player to claim their bounty\n`
         + `/${Commands.togglelifetags} - Toggles if you are tagged whenever you are done working or completed your sentence`;
+        + `/${Commands.communityService} - Perform community service to reduce your bounty\n`;
```
you forgot to add this command to the Commands.ts file

MaikoKingma commented on plugin.ts:
```diff
@@ -58,8 +58,9 @@ export class Plugin extends AbstractPlugin {
     const bountiesCommand = new BotCommand([Commands.bounties], "", this.bounties.bind(this), false);
     const placeBountyCommand = new BotCommand([Commands.placebounty], "", this.placeBounty.bind(this), false);
     const killPlayerCommand = new BotCommand([Commands.kill], "", this.kill.bind(this), false);
+    const communityServiceCommand = new BotCommand([Commands.cs1, Commands.cs2], "", this.communityService.bind(this), false);
```
you forgot to add these commands to the Commands.ts file

MaikoKingma commented on plugin.ts:
```diff
@@ -401,4 +402,50 @@ export class Plugin extends AbstractPlugin {
+    }
+    const multiplier: number = chat.getSetting(Strings.CS_MULTIPLIER_SETTING);
+
+    lifeUser.startCommunityService(minutes, () => {
```
lifeChatData, targetUser and bounty are not defined in this function

TomChermin committed:
Fixed CS bugs

TomChermin committed:
Fixed minor (function breaking) bugs

Added cs_multiplier
Added /n in /Life

TomChermin committed:
Update plugin.ts

Fixed the undefined targetUser bug (maybe) and it should no longer be possible to do cs without a bounty (I hope)

MaikoKingma commented on plugin.ts:
```diff
@@ -401,4 +403,53 @@ export class Plugin extends AbstractPlugin {
+    const multiplier: number = chat.getSetting(Strings.CS_MULTIPLIER_SETTING);
+
+    lifeUser.startCommunityService(minutes, () => {
+     const parameters = match.split(' ');
```
`match` is undefined here when a user does `/cs` without tagging another user (eg. applying the cs on himself)

MaikoKingma replied:
sidenote, I'm guessing the `match` variable is expected to contains some kind of user tag? I have never used `match` and don't know whats its for.

TomChermin replied:
I've also never used match and don't know what it's for :P

TomChermin committed:
Match & ScoreToGain fix

Removed match in cs and redefined ScoreToGain

TomChermin commented:
Move bounty check to the start of cs instead of the resolve

TomChermin commented:
Add a status for cs

Agadar reviewed (commented):
Let me know when this is fully tested.

TomChermin committed:
Minor fixes

Added full status support and fixed the 0 bounty bug

MaikoKingma commented on plugin.ts:
```diff
@@ -401,4 +412,55 @@ export class Plugin extends AbstractPlugin {
+    const multiplier: number = chat.getSetting(Strings.CS_MULTIPLIER_SETTING);
+
+    lifeUser.startCommunityService(minutes, () => { //The actual community service
+     const lifeChatData = this.helper.getOrCreateLifeChatsData(chat.id);
```
this line is needed before line 418 otherwise lifeChatData is undefined there

TomChermin committed:
Update plugin.ts

I think I fix, maybe I fuck

TomChermin committed:
"Final" update

I think it works completely now

TomChermin committed:
Even finaller push/commit/pull/whatever

Now with accurate end text!

MaikoKingma commented on plugin.ts:
```diff
@@ -401,4 +412,54 @@ export class Plugin extends AbstractPlugin {
+
+      //Reduces the bounty by x amount
+      scoreToGain += lifeUser.occupation!.waitingTime * 20 * multiplier;
+      chatBounty.bounty += scoreToGain * (-1);
```
`scoreToGain = scoreToGain > chatBounty ? chatBounty : scoreToGain`

TomChermin committed:
Just work

Please just work

TomChermin committed:
Ja weer een update

Nu moet ie werken (ook al doet ie het waarschijnlijk niet)

TomChermin committed:
Added highway to /life

See title

Agadar commented:
What's the status on this @TomChermin @MaikoKingma ?
